test/unit/common/services/scalingo-client_test.js

```diff
@@ -134,6 +134,54 @@ describe('Scalingo client', () => {
     });
   });
 
+  describe('#ScalingoClient.deployFromSCM', () => {
+    let manualDeployStub;
+    let scalingoClient;
+
+    beforeEach(async function () {
+      manualDeployStub = sinon.stub();
+      sinon.stub(scalingo, 'clientFromToken').resolves({ SCMRepoLinks: { manualDeploy: manualDeployStub } });
+
+      scalingoClient = await ScalingoClient.getInstance('production');
+    });
+
+    it('should not deploy without app given', async () => {
+      try {
+        await scalingoClient.deployFromSCM(null, 'v1.0');
+        expect.fail('Should throw an error when no application given');
+      } catch (e) {
+        expect(e.message).to.equal('No application to deploy.');
+      }
+    });
+
+    it('should not deploy without a release tag given', async () => {
+      try {
+        await scalingoClient.deployFromSCM('pix-app', null);
+        expect.fail('Should throw an error when no release tag given');
+      } catch (e) {
+        expect(e.message).to.equal('No release tag to deploy.');
+      }
+    });
+
+    it('should deploy an application for a given tag', async () => {
+      await scalingoClient.deployFromSCM('pix-app-production', 'v1.0');
+      sinon.assert.calledWithExactly(manualDeployStub, 'pix-app-production', 'v1.0');
```

Comment:
expect(manualDeployStub).to.have.been.calledOnceWithExactly('pix-app-production', 'v1.0');

Reply:
Remplacé ✅